### ISSUE TITLE
docs: navigation sidebar depth is too small

### ISSUE THREAD

**Issue by michoecho** [scylladb/scylladb#11548]
Commit 0c2a18af2d50271ab1cf5cd8392dbe54d8bb7791 shrunk the depth of the navigation sidebar on docs.scylladb.com to at most 3. Why? Now when I go to https://docs.scylladb.com/stable/architecture/sstable/sstable3/sstables-3-summary.html, I can't switch from SSTable Summary File docs to SSTable Index file docs without going through the containing https://docs.scylladb.com/stable/architecture/sstable/sstable3/index.html# page.

Suggested fix: increase the depth to at least 4.

**Comment by tzach** [comment]
@michoecho can you please add relevant screen capture and screen width?
This kind of problems is very much setup related.

**Comment by michoecho** [comment]
> screen width?

No, I don't mean the CSS. Setup doesn't matter here — it's about the logical depth of the navigator.

Before 0c2a18af2d50271ab1cf5cd8392dbe54d8bb7791:
![2022-09-15T12:28:05+02:00](https://user-images.githubusercontent.com/45052050/190381854-5cea4744-f8c2-4d46-be99-9af9fbd5f58a.png)
After 0c2a18af2d50271ab1cf5cd8392dbe54d8bb7791:
![2022-09-15T12:28:15+02:00](https://user-images.githubusercontent.com/45052050/190381966-4a901775-ab7f-4dee-a6cb-d8afbf8817b5.png)

Entries deeper than 3 levels are not shown in the sidebar.

**Comment by tzach** [comment]
Thanks, I understand now.
We should avoid 3 level nesting. In this case we can move sstable2 and sstable3 one level up in the hierarchy
@annastuchlik what do you think?

**Comment by nyh** [comment]
I think we should move all the children of sstable3 one level up, and keep "SSTable 2.0" where it is, so users will only see sstable 3 unless they go into sstable 2. I.e., something like this:
```
SSTable
   + SSTable Data File Format
   + SSTable Statistics File Format
   + ...
   + SSTable 2.0 Format (deprecated)
```

The SSTable 2.0 format reference is very interesting for developers and for historical reasons (so please don't just remove it), but nobody is using this format any more on new installations, so most of the time, it would not the right document to look at.

The only problem with moving stuff around the URL tree is the loss of Google search ranking...

**Comment by michoecho** [comment]
Another case of deep (level 5) nesting are the upgrade guides, e.g. `ScyllaDB Open Source / Scylla for Administrators / Upgrade Scylla / Upgrade ScyllaDB Open Source / Upgrade Guide - ScyllaDB 5.x Maintenance Release / Upgrade Guide - ScyllaDB 5.x.y to 5.x.z for Ubuntu 18.04, 20.04, and 22.04`.

I think this case would also be better with a level 4 sidebar instead of a level 3 sidebar — but it's just my opinion.

**Comment by annastuchlik** [comment]
@michoecho Thanks for your feedback! I agree with you that there some pages are hard to find because they don't appear on the navigation sidebar. I even opened a similar issue to this one, but it turned out that there are technical reasons why we shouldn't increase the depth of the sidebar navigation - deeper levels increased the documentation build time to an unacceptable level.

Our current approach is to gradually change the structure of the docs so that the pages appear on the sidebar. For example, when I add a new page (like [this](https://docs.scylladb.com/stable/using-scylla/workload-attributes.html) one), I add it higher in the page tree.  
We also try to restructure the existing docs. For example, we moved the CQL reference page higher in the hierarchy. This, however, comes with a price, e.g. such a change makes a page harder to find via Google.

I'm leaving this issue open to introduce the structure changes suggested by @nyh.

**Comment by michoecho** [comment]
> deeper levels increased the documentation build time to an unacceptable level.

This can't be true. The difference between leveI just compiled with both depth 3 and depth 4 (Sphinx default, which was reduced to 3 in 0c2a18af2d50271ab1cf5cd8392dbe54d8bb7791), and there was no visible difference in build times. Even with depth 5 (which includes all the update guides), the difference is ~20%.

You can save much more than that by parallelizing the build. When I tried to build the docs to report this issue, I was perplexed that the docs build longer than Scylla does, so I did some digging and found out that we don't used parallel builds (they were disabled scylladb/sphinx-scylladb-theme@eaab307276a272c74397f552781ed607e53a9872) because one of the used Sphinx extensions, `sphinx-sitemap` (Is it even used? https://docs.scylladb.com/sitemap.xml is 404) inhibits parallelism. So I did more digging and it turns out that parallelizing this extension is relatively straightforward.

Feel free to replace sphinx-sitemap with my parallelized fork (https://github.com/michoecho/sphinx-sitemap/, you can add github (non-PyPI) dependencies in Poetry) and add `-jauto` to the Sphinx opts, to get parallelism back.

**Comment by annastuchlik** [comment]
@dgarcia360 Can we discuss the solution proposed by @michoecho ?

**Comment by dgarcia360** [comment]
Thanks @michoecho,

We had some issues with extensions that did not support parallel builds, and also our custom toctree it increased the build time by a lot when there were more than 3 level as described in https://github.com/scylladb/scylla-docs/pull/3676#issuecomment-911340038

I'll test the solution proposed replacing the extension and enabling back paralell builds. If it works, we should be able to remove the restriction.

**Comment by tzach** [comment]
Putting a side the compile time issue, deep nesting does not looks good we are moving away from it.
